modules/core/src/main/scala/org/tessellation/infrastructure/trust/storage/TrustStorage.scala

```diff
@@ -4,49 +4,120 @@ import cats.Monad
 import cats.effect.Ref
 import cats.syntax.flatMap._
 import cats.syntax.functor._
-import cats.syntax.functorFilter._
 import cats.syntax.option._
+import cats.syntax.partialOrder._
 
-import org.tessellation.domain.trust.storage.TrustStorage
+import org.tessellation.cli.method.ordinalTrustUpdateDelay
+import org.tessellation.domain.trust.storage._
+import org.tessellation.schema.SnapshotOrdinal
 import org.tessellation.schema.peer.PeerId
-import org.tessellation.schema.trust.{PeerObservationAdjustmentUpdateBatch, PublicTrust, TrustInfo}
+import org.tessellation.schema.trust._
+
+import derevo.circe.magnolia.encoder
+import derevo.derive
+import monocle.Monocle.toAppliedFocusOps
 
 object TrustStorage {
 
-  def make[F[_]: Monad: Ref.Make]: F[TrustStorage[F]] = Ref[F].of[Map[PeerId, TrustInfo]](Map.empty).map(make(_))
+  @derive(encoder)
+  case class TrustStore(
+    trust: TrustMap,
+    currentOrdinalTrust: OrdinalTrustMap,
+    nextOrdinalTrust: OrdinalTrustMap
+  )
+
+  def make[F[_]: Monad: Ref.Make]: F[TrustStorage[F]] = make(TrustMap.empty)
 
-  def make[F[_]: Monad: Ref.Make](trustUpdates: Option[PeerObservationAdjustmentUpdateBatch]): F[TrustStorage[F]] =
-    trustUpdates.fold(make)(trust => make.flatTap(_.updateTrust(trust)))
+  def make[F[_]: Monad: Ref.Make](
+    trustInfo: TrustMap
+  ): F[TrustStorage[F]] =
+    Ref[F]
+      .of[TrustStore](TrustStore(trustInfo, OrdinalTrustMap.empty, OrdinalTrustMap.initial))
+      .map(make(_))
 
-  def make[F[_]: Monad](trust: Ref[F, Map[PeerId, TrustInfo]]): TrustStorage[F] =
+  def make[F[_]: Monad: Ref.Make](
+    trustUpdates: Option[PeerObservationAdjustmentUpdateBatch]
+  ): F[TrustStorage[F]] =
+    trustUpdates.fold(make)(batch => make.flatTap(_.updateTrust(batch)))
+
+  def make[F[_]: Monad](trustStoreRef: Ref[F, TrustStore]): TrustStorage[F] =
     new TrustStorage[F] {
 
       def updateTrust(trustUpdates: PeerObservationAdjustmentUpdateBatch): F[Unit] =
-        trust.update { t =>
-          t ++ trustUpdates.updates.map { trustUpdate =>
-            trustUpdate.id -> t.getOrElse(trustUpdate.id, TrustInfo()).copy(trustLabel = trustUpdate.trust.value.some)
+        trustStoreRef.update { store =>
+          val t = store.trust
+          val updated = t.trust ++ trustUpdates.updates.map { trustUpdate =>
+            trustUpdate.id -> t.trust
+              .getOrElse(trustUpdate.id, TrustInfo())
+              .focus(_.trustLabel)
+              .replace(trustUpdate.trust.value.some)
           }.toMap
+          store.focus(_.trust.trust).replace(updated)
+        }
+
+      def updatePredictedTrust(trustUpdates: Map[PeerId, Double]): F[Unit] =
+        trustStoreRef.update { store =>
+          val t = store.trust
+          val updated = t.trust ++ trustUpdates.map {
+            case (k, v) =>
+              k -> t.trust.getOrElse(k, TrustInfo()).focus(_.predictedTrust).replace(v.some)
+          }
+          store.focus(_.trust.trust).replace(updated)
```

Comment:
Why is `updateTrust` and `updatePredictedTrust` updating the same value in store but they are called in a way as if they were updating separate values? It's like stored trust is a trust and predicted trust at the same time.

Reply:
`updateTrust` updates the trust labels [(link](https://github.com/cngo-github/tessellation/blob/PROT-238-20230626/modules/core/src/main/scala/org/tessellation/infrastructure/trust/storage/TrustStorage.scala#L52)). The trust labels are trust values provided by the user. The labels are used to help determine the predicted trust values.

`updatePredictedTrust` updates the predicted trust values ([link](https://github.com/cngo-github/tessellation/blob/PROT-238-20230626/modules/core/src/main/scala/org/tessellation/infrastructure/trust/storage/TrustStorage.scala#L63)). The predicted trust values are the values used as the trustworthiness of a peer.

Reply:
I think this is resolved? These are separate fields being updated for different purposes.